### PR TITLE
chore: add registry dialog

### PR DESCRIPTION
**Pull request by deboer-tim**
### What does this PR do?

First part of #8103. Switches adding a new registry from a row at the bottom of the table to a dialog, as per the design.

Keeping this relatively simple for the first step. Future PRs could modify the dialog to support editing as well.

### Screenshot / video of UI

Screenshot 2024-08-08 at 1 52 16 PM

### What issues does this PR fix or reference?

First part of #8103.

### How to test this PR?

Add a new registry.

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by axel7083 — APPROVED** [review]
LGTM, great work ! Tested and works fine

File: packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte [hunk 1]
```diff
@@ -438,46 +439,51 @@ function removeExistingRegistry(registry: containerDesktopAPI.Registry) {
...
+    </div>
+    <svelte:fragment slot="validation"
+      ><ErrorMessage error={errorResponses.find(o => o.serverUrl === newRegistryRequest.serverUrl)?.error ?? ''}
+      ></ErrorMessage
+      ></svelte:fragment>
+    <svelte:fragment slot="buttons">
+      <Button type="link" on:click={() => (showNewRegistryForm = false)}>Cancel</Button>
+      <Button
+        type="primary"
+        disabled={!newRegistryRequest.serverUrl || !newRegistryRequest.username || !newRegistryRequest.secret}
```
Comment (dgolovin) [inline packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte]: nit:
```suggestion
        disabled={!newRegistryRequest.serverUrl.trim() || !newRegistryRequest.username.trim() || !newRegistryRequest.secret.trim()}
```

**Review by dgolovin — APPROVED** [review]
LGTM.
nit: trim call required for disabled  attribute, right now it would let create registry even if I put spaces in every field.

**Review by gastoner — APPROVED** [review]
I tried it out, and it works fine!

File: packages/renderer/src/lib/ui/PasswordInput.svelte [hunk 2]
```diff
@@ -30,7 +30,7 @@ async function onShowHide() {
   class={$$props.class || ''}
   id="password-{id}"
   name="password-{id}"
-  placeholder="Password"
+  placeholder="password"
```
Comment (benoitf) [inline packages/renderer/src/lib/ui/PasswordInput.svelte:33]: looks like unrelated

**Review by benoitf — COMMENTED** [review]
image


should trim whitespaces

I think I would expect `Add` button rather than `Login` button

title is 'Add a registry' so I expect to `add` it rather than `login`

image

File: packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte (hunk 1 above)
Comment (deboer-tim) [reply packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte]: Included in new commit (lint reformatted).

**Comment by deboer-tim** [comment]
New commit:
- Renames button to Add. (I had done this initially and then 'fixed' it to match the design, but this is better)
- Trims whitespace when checking if the button should be enabled.
- Automatically trims whitespace when adding (so if you accidentally type or copy/paste ' https//quay.io ' it is fixed).

File: packages/renderer/src/lib/ui/PasswordInput.svelte (hunk 2 above)
Comment (deboer-tim) [reply packages/renderer/src/lib/ui/PasswordInput.svelte:33]: Password is a shared component that had a capitalized Password placeholder, but it is only used on this page so far and the design has lowercase placeholders for username and password so I changed it here. I don't have a strong opinion (other than the cases should match) but lower looked slightly better to me.
